app/src/main/java/teamagam/ConnectionTest/HttpValidator.java

```diff
@@ -0,0 +1,40 @@
+package teamagam.ConnectionTest;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+
+public class HttpValidator implements ValidatorAsyncTask.Validator {
+
+    private static final String URL_PREFIX = "http://";
+
+    @Override
+    public boolean validate(String url) {
+        try {
+            HttpURLConnection connection = (HttpURLConnection) parseUrl(url).openConnection();
```

Comment:
Ideally I would ask you to extract method "establishConnection" that throws exception when code isn't HTTP_OK, but never mind now.